src/tools/illink/src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -1167,7 +1167,7 @@ GenericParameterValue genericParam
 
 			// For now, if the intrinsic doesn't set a return value, fall back on the annotations.
 			// Note that this will be DynamicallyAccessedMembers.None for the intrinsics which don't return types.
-			returnValue ??= calledMethod.ReturnsVoid () ? MultiValueLattice.Top : annotatedMethodReturnValue;
+			returnValue ??= (calledMethod.ReturnsVoid () && !calledMethod.IsConstructor ()) ? MultiValueLattice.Top : annotatedMethodReturnValue;
```

Comment:
One thing I struggled with when I was thinking about just fixing this in #101031 was whether it would be a problem to consider calls to constructors to always be `newobj`. Constructors will be called with `call` instead of `newobj` sometimes (when calling the base constructor from a derived type constructor) and in those cases they "return void" and `.Top` would be still the correct answer.

Maybe it's fine to do this approximation, but wanted to raise this in case you could see problems with that.